src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4901,6 +4901,11 @@ private void VisitDeconstructionArguments(ArrayBuilder<DeconstructionVariable> v
                     else
                     {
                         var targetType = variable.Type;
+                        if (variable.Expression.ExpressionSymbol is SourceLocalSymbol local && local.IsVar && rightPart is BoundFieldAccess rightField)
```

Comment:
>rightPart is BoundFieldAccess rightField [](start = 110, length = 40)

Will this handle an explicit field access on the RHS?
```C#
    static void F(object? x1, object y1)
    {
        if (x1 == null) return;
        y1 = null; // 1
        var t = (x1, y1);
        var (x2, y2) = (t.Item1, t.Item2);
        x2 = null; // 2
        y2 = null;
    }
```

Reply:
Added a test to cover this